text/0000-rust-governance.md

```diff
@@ -0,0 +1,728 @@
+the fact that subteam leaders will thus remain in close communication on
+cross-cutting concerns in particular -- serves to greatly mitigate that risk.
+
+As with any change to governance, there is risk that this RFC would harm
+processes that are working well. In particular, bringing on a large number of
+new people into official decision-making roles carries a risk of culture clash
+or problems with consensus-building.
+
+By setting up this change as a relatively slow build-out from the current core
+team, some of this risk is mitigated: it's not a radical restructuring, but
+rather a refinement of the current process. In particular, today core team
+members routinely seek input directly from other community members who would be
+likely subteam members; in some ways, this RFC just makes that process more
+official.
+
+For the moderation subteam, there is a significant shift toward strong
+enforcement of the CoC, and with that a risk of *over*-application: the goal is
+to make discourse safe and productive, not to introduce fear of violating the
+CoC. The moderation guidelines, careful selection of moderators, and ability to
+withdraw moderators mitigate this risk.
+
+# Alternatives
+
+There are numerous other forms of open-source governance out there, far more
+than we can list or detail here. And in any case, this RFC is intended as an
+expansion of Rust's existing governance to address a few scaling problems,
+rather than a complete rethink.
+
+[Mozilla's module system][module], was a partial inspiration for this RFC. The
+proposal here can be seen as an evolution of the module system where the subteam
+leaders (module owners) are integrated into an explicit core team, providing for
+tighter intercommunication and a more unified sense of vision and purpose.
+Alternatively, the proposal is an evolution of the current core team structure
+to include subteams.
+
+One seemingly minor, but actually important aspect is *naming*:
+
+* The name "subteam" (from [jQuery][jq]) felt like a better fit than "module" both
+to avoid confusion (having two different kinds of modules associated with
+Mozilla seems problematic) and because it emphasizes the more unified nature of
+this setup.
+
+* The term "leader" was chosen to reflect that there is a vision for each subteam
+(as part of the larger vision for Rust), which the leader is responsible for
+moving the subteam toward. Notably, this is how "module owner" is actually
+defined in Mozilla's module system:
+
+  > A "module owner" is the person to whom leadership of a module's work has been
+  > delegated.
+
+* The term "team member" is just following standard parlance. It could be
+replaced by something like "peer" (following the module system tradition), or
+some other term that is less bland than "member". Ideally, the term would
+highlight the significant stature of team membership: being part of the
+decision-making group for a substantial area of the Rust project.
```

Comment:
In my ideal world 'peer' means being a peer for the project as a whole and the core-team are also peers with a particular responsibility towards strategic/global issues, rather than 'peer' being part of a sub-group.